src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -165,80 +177,66 @@ export default class AmalgamatingBusinesses extends Mixins(CommonMixin) {
   // Local properties
   initialBusinessLookupObject = EmptyBusinessLookup
   businessTableValid = false
+  snackbar = false
+  snackbarText = ''
 
   // Button properties
   isAddingAmalgamatingBusiness = false
   isAddingAmalgamatingForeignBusiness = false
 
-  readonly BusinessLookupServices = BusinessLookupServices
-
-  // Cancel button in "Add an Amalgamating Business" is pressed.
-  addAmalgamatingBusinessCancel (): void {
-    this.isAddingAmalgamatingBusiness = false
-    this.setAmalgamatingBusinessesValid(true)
-  }
-
-  // "Add an Amalgamating Business" button is pressed.
-  onAddBusinessClick (): void {
-    this.isAddingAmalgamatingBusiness = true
-    this.isAddingAmalgamatingForeignBusiness = false
-    this.setAmalgamatingBusinessesValid(false)
-  }
-
-  // "Add an Amalgamating Foreign Business" button is pressed.
-  onAddForeignBusinessClick (): void {
-    this.isAddingAmalgamatingBusiness = false
-    this.isAddingAmalgamatingForeignBusiness = true
-    this.setAmalgamatingBusinessesValid(false)
-  }
-
   async saveAmalgamatingBusiness (businessLookup: BusinessLookupIF): Promise<void> {
-    let business = null
-
-    // Get the amalgamating business information, mailing address, and email if in LEAR.
+    // Get the amalgamating business information, auth info and addresses, and email if in LEAR.
     // Otherwise, return the businesslookup object.
     const data = await Promise.all([
       LegalServices.fetchBusinessInfo(businessLookup.identifier),
       AuthServices.fetchAuthInfo(businessLookup.identifier),
-      LegalServices.fetchAddresses(businessLookup.identifier)
-    ]).catch((error) => {
-      return error
-    })
+      LegalServices.fetchAddresses(businessLookup.identifier),
+      LegalServices.fetchFilings(businessLookup.identifier)
+    ]).catch(() => {}) // ignore errors
 
-    if (data.length === 3) {
-      business = data[0].data?.business
-      business.businessContact = data[1].contacts[0]
-      business.officeAddress = data[2]
+    // check for errors
+    if (!data || data.length !== 4) {
+      this.snackbarText = 'Unable to add this business.'
+      this.snackbar = true
+      return
```

Comment:
I think I need to rework this as well, since it doesn't handle non-affiliated businesses.

Reply:
Fixed.